tests/Credentials/CredentialProviderTest.php

```diff
@@ -664,6 +664,167 @@ public function testEnsuresSourceProfileIsSpecified()
         }
     }
 
+    public function testAssumeRoleInConfigFromCredentialsSourceFails()
+    {
+        $dir = $this->clearEnv();
+        putenv(CredentialProvider::ENV_KEY . '=abc');
+        putenv(CredentialProvider::ENV_SESSION . '');
+
+        $credentials = <<<EOT
+[assume]
+role_arn = arn:aws:iam::012345678910:role/role_name
+credential_source = Environment
+EOT;
+        file_put_contents($dir . '/credentials', $credentials);
+        putenv('HOME=' . dirname($dir));
+
+        try {
+            $result = CredentialProvider::getAssumeRoleCredentials(
+                'assume',
+                'assume',
+                $dir . '/credentials',
+                []
+            );
+            self::assertInstanceOf('GuzzleHttp\Promise\RejectedPromise', $result);
```

Comment:
Is there any way to test this rejected promise is specifically the env rejection we're looking for?

Reply:
> Is there any way to test this rejected promise is specifically the env rejection we're looking for?

Bump